examples/dmd/de_dg_advection_greedy.cpp

```diff
@@ -0,0 +1,1613 @@
+                fout.open("parameters.txt", std::ios::app);
+                fout << to_string(f_factor) << std::endl;
+                fout.close();
+            }
+        }
+
+        if (true_solution_u == NULL)
+        {
+
+            true_solution_u = new Vector(U->Size());
+            *true_solution_u = U->GetData();
+            tot_true_solution_u_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                            *true_solution_u, *true_solution_u));
+
+        }
+
+        CAROM::Vector* result_U = dmd_U->predict(t_final);
+
+
+        Vector dmd_solution_U(result_U->getData(),
+                              result_U->dim()); // Potential Problem
+        Vector diff_u(true_solution_u->Size());
+        subtract(dmd_solution_U, *true_solution_u, diff_u);
+
+
+        double tot_diff_norm_u = sqrt(InnerProduct(MPI_COMM_WORLD, diff_u, diff_u));
+        rel_diff = tot_diff_norm_u / tot_true_solution_u_norm;
+
+
+        if (myid == 0)
+        {
+            std::cout << "Rel. diff. of DMD temp. (u) at t_final at f_factor " << f_factor
+                      << ": "
+                      << rel_diff << std::endl;
+        }
+
+        delete result_U;
+
+        if (myid == 0)
+        {
+            printf("Elapsed time for training DMD: %e second\n",
+                   dmd_training_timer.RealTime());
+        }
+    }
+    else if (online)
+    {
+        CAROM::Vector* result_U = dmd_U->predict(ts[0]);
+        Vector initial_dmd_solution_U(result_U->getData(),
+                                      result_U->dim());  // Potential Problem
+        u_gf->SetFromTrueDofs(initial_dmd_solution_U);
+
+        VisItDataCollection dmd_visit_dc("DMD_DG_Advection_Greedy_"
+                                         +
+                                         to_string(f_factor), pmesh);
+        dmd_visit_dc.RegisterField("temperature", u_gf);
+        if (visit)
+        {
+            dmd_visit_dc.SetCycle(0);
+            dmd_visit_dc.SetTime(0.0);
+            dmd_visit_dc.Save();
+        }
+
+        delete result_U;
+
+        if (visit)
+        {
+            for (int i = 1; i < ts.size(); i++)
+            {
+                if (i == ts.size() - 1 || (i % vis_steps) == 0)
+                {
+                    result_U = dmd_U->predict(ts[i]);
+                    if (myid == 0)
+                    {
+                        std::cout << "Predicting temperature using DMD at: " << ts[i] << std::endl;
+                    }
+
+                    Vector dmd_solution_U(result_U->getData(),
+                                          result_U->dim());  // Potential problem
+                    u_gf->SetFromTrueDofs(dmd_solution_U);
+
+                    dmd_visit_dc.SetCycle(i);
+                    dmd_visit_dc.SetTime(ts[i]);
+                    dmd_visit_dc.Save();
+
+                    delete result_U;
+                }
+            }
+        }
+
+        dmd_prediction_timer.Stop();
```

Comment:
When running the example commands I noticed the timer printout for the second and third commands was 0.0. I think this `dmd_prediction_timer` might be missing a `.Start()` in this section?

Reply:
Added the correct timer.